### DISCUSSION TITLE
How to configure the Microsoft Search option to connect to external content

### DISCUSSION THREAD

**Discussion by saregbesola** [microsoft-search/pnp-modern-search#2176] [Q&A]
Can anyone please help with how to configure PnP search web part to use External Items in the "entity types to search" when using Microsoft Search as a DataSource? I'm trying to see how I can surface content from an external system in the search result but I haven't find any documentation on that. Any help will be appreciated. @wobba @FranckyC -kindly help

**Comment by wobba** [comment]
See https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/#microsoft-search, and ensure you go to https://tenant-admin.sharepoint.com/_layouts/15/online/AdminHome.aspx#/webApiPermissionManagement and grant the needed permissions for the API.

**Comment by gretchunkim** [comment]
The key to configuring external content (MS Graph  Connector) and PnP Search Results is utilization of **Connect Sources** field on the 1 of 4 page.  Type the IDs (connector name here) , ensure that you select **Entity types** to search "External Items" then type applicable files in the "Selected fields" section on page 1 of 4 (ver4.8.0).  Use Debug under the Layouts to see all results. You shouldn't see error here if configured correctly. If permission issue you should see something here.
